komodoenv/update.py

```diff
@@ -409,9 +409,15 @@ def create_pth(config: dict, srcpath: Path, dstpath: Path):
         / "lib"
         / ("python" + config["python-version"])
         / "site-packages"
-        / "_komodo.pth"
     )
-    with open(path, "w", encoding="utf-8") as f:
+    # If upgrading from an old komodoenv using '_komodo.pth'
+    # to a newer which uses 'zzz_komodo.pth' we must make sure to
+    # remove the old _komodo.pth.
+    old_style_pth = path / "_komodo.pth"
+    if old_style_pth.is_file():
+        old_style_pth.unlink()
+    new_style_pth = path / "zzz_komodo.pth"
+    with open(new_style_pth, "w", encoding="utf-8") as f:
```

Comment:
```suggestion
    with contextlib.suppress(FileNotFoundError):
        (path / "_komodo.pth").unlink()
    with open(path / "zzz_komodo.pth", "w", encoding="utf-8") as f:
```

Unlinking the file and ignoring the error is technically better as it is an atomic operation and isn't prone to race conditions like the original code. If the file is deleted between `.is_file()` check and `.unlink()` then you'll get a `FileNotFound` and crash the program. The odds of this happening are miniscule of course.

IMO, it's good style to try deleting the file and checking the error as it's technically faster (negligibly: 1 syscall vs 2, and the OS still has to check whether the file exists anyway so why should we also do it) and has no downsides.